kernel/src/syscall/pidfd_open.rs

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::{
+    fs::{file_table::FdFlags, utils::StatusFlags},
+    prelude::*,
+    process::{process_table, Pid, PidFile},
+    syscall::SyscallReturn,
+};
+
+pub fn sys_pidfd_open(pid: Pid, flags: u32, ctx: &Context) -> Result<SyscallReturn> {
+    let is_nonblocking = {
+        let flags = PidfdFlags::from_bits(flags)
+            .ok_or_else(|| Error::with_message(Errno::EINVAL, "invalid flags"))?;
+        debug!("pid = {}, flags = {:?}", pid, flags);
+        flags.contains(PidfdFlags::PIDFD_NONBLOCK)
+    };
+
+    if pid.cast_signed() < 0 {
+        return_errno_with_message!(Errno::EINVAL, "negative PIDs or PGIDs are not valid");
```

Comment:
I think this has nothing to do with PGIDs? Maybe just say `negative PIDs are not valid`.